external-crates/move/crates/move-compiler/tests/move_2024/ide_mode/missing_type_suggestions.ide.exp

```diff
@@ -0,0 +1,117 @@
+
+warning[W09009]: unused struct field
+  ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:2:26
+  │
+2 │     public struct S<T> { x: T }
+  │                          ^ The 'x' field of the 'S' type is unused
+  │
+  = This warning can be suppressed with '#[allow(unused_field)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
+
+note[I15006]: IDE path autocomplete
+  ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:2:29
+  │
+2 │     public struct S<T> { x: T }
+  │                             ^ Possible in-scope names: 'A -> a::m::A', 'Option -> std::option::Option', 'S -> a::m::S', or 'T'
+
+note[I15006]: IDE path autocomplete
+   ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:6:1
+   │  
+ 6 │ ╭ module a::test {
+ 7 │ │     use a::m::{Self, S, A};
+ 8 │ │ 
+ 9 │ │     public fun p(a:
+10 │ │ 
+11 │ │     public fun q():
+12 │ │ }
+   │ ╰─^ Possible in-scope names: 'A -> a::m::A', 'Option -> std::option::Option', 'S -> a::m::S', 'Self -> a::test', 'm -> a::m', 'option -> std::option', 'vector -> std::vector', 'B -> 0x42', 'K -> 0x19', 'M -> 0x40', 'a -> 0x44', 'b -> 0x45', 'k -> 0x19', 'std -> 0x1', or 'sui -> 0x2'
+
+warning[W09001]: unused alias
+  ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:7:16
+  │
+7 │     use a::m::{Self, S, A};
+  │                ^^^^ Unused 'use' of alias 'm'. Consider removing it
+  │
+  = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
+
+warning[W09001]: unused alias
+  ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:7:22
+  │
+7 │     use a::m::{Self, S, A};
+  │                      ^ Unused 'use' of alias 'S'. Consider removing it
+  │
+  = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
+
+warning[W09001]: unused alias
+  ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:7:25
+  │
+7 │     use a::m::{Self, S, A};
+  │                         ^ Unused 'use' of alias 'A'. Consider removing it
+  │
+  = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
+
+warning[W09002]: unused variable
+  ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:9:18
+  │
+9 │     public fun p(a:
+  │                  ^ Unused parameter 'a'. Consider removing or prefixing with an underscore: '_a'
+  │
+  = This warning can be suppressed with '#[allow(unused_variable)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
+
+note[I15006]: IDE path autocomplete
+   ┌─ tests/move_2024/ide_mode/missing_type_suggestions.move:11:5
+   │  
+ 9 │ │     public fun p(a:
```

Comment:
It might be good to include primitive types for places where we know we are expecting a type. We could do it at the IDE level but I am not sure how easy it will be to tell then if we are in a position where a type is expected or perhaps a different member of a module

Reply:
Actually, on the second thought it shouldn't be difficult on the IDE side as we can tell when the chain is part of a type or part of an expression